spring-boot-starter-elasticsearch-evolution/src/main/java/com/senacor/elasticsearch/evolution/spring/boot/starter/autoconfigure/ElasticsearchEvolutionAutoConfiguration.java

```diff
@@ -55,18 +64,59 @@ public ElasticsearchEvolutionInitializer elasticsearchEvolutionInitializer(Elast
     @ConditionalOnClass(RestHighLevelClient.class)
     public static class RestHighLevelClientConfiguration {
 
+        @Value("${spring.elasticsearch.rest.username:}")
+        private String username ;
+
+        @Value("${spring.elasticsearch.rest.password:}")
+        private String password;
+
+        @Value("${spring.elasticsearch.rest.connectTimeout:5000}")
+        private int connectTimeout ;
+
+        @Value("${spring.elasticsearch.rest.socketTimeout:300000}")
+        private int socketTimeout ;
+
+        @Value("${spring.elasticsearch.rest.connectionRequestTimeout:3000}")
+        private int connectionRequestTimeout;
+
+        @Value("${spring.elasticsearch.rest.maxRetryTimeoutMillis:300000}")
+        private int maxRetryTimeoutMillis;
+
         /**
          * @return default RestHighLevelClient if {@link org.springframework.boot.autoconfigure.elasticsearch.ElasticsearchRestClientAutoConfiguration} is not used
          * and no RestHighLevelClient is available.
          */
         @Bean
         @ConditionalOnMissingBean
         public RestHighLevelClient restHighLevelClient(@Value("${spring.elasticsearch.rest.uris:http://localhost:9200}") String... uris) {
```

Comment:
In this case you can just create your own RestHighLevelClient Bean and then the autoconfig from elasticsearch-evolution will not apply.
So we will not merge this PR.

Reply:
Thank you for your reply！